library/ssl_cli.c

```diff
@@ -136,18 +136,18 @@ static int ssl_write_hostname_ext( mbedtls_ssl_context *ssl,
      * } ServerNameList;
      *
      */
-    *p++ = (unsigned char)( ( MBEDTLS_TLS_EXT_SERVERNAME >> 8 ) & 0xFF );
-    *p++ = (unsigned char)( ( MBEDTLS_TLS_EXT_SERVERNAME      ) & 0xFF );
+    *p++ = MBEDTLS_BYTE_1( MBEDTLS_TLS_EXT_SERVERNAME );
+    *p++ = MBEDTLS_BYTE_0( MBEDTLS_TLS_EXT_SERVERNAME );
```

Comment:
Why not use `MBEDTLS_PUT_UINT16_BE()` here? It would seem neater, but I may be missing something. (Also in lns 142-143, 145-146, 149-150 etc.)

Reply:
I suppose `p` needs to be updated.

Reply:
Could do something like:
```C
  MBEDTLS_PUT_UINT16_BE(..
  p += 2;
```
But it's not obvious which is easier to read.